plaso/storage/redis/merge_reader.py

```diff
@@ -0,0 +1,292 @@
+  }
+
+  def __init__(self, storage_writer, task, redis_client=None):
+    """Initializes a redis storage merge reader.
+
+    Args:
+      storage_writer (StorageWriter): storage writer.
+      task (Task): the task whose store is being merged.
+      redis_client (Optional[Redis]): redis client to query. If specified, no
+          new client will be created.
+
+    Raises:
+      RuntimeError: if an add container method is missing.
+    """
+    super(RedisMergeReader, self).__init__(storage_writer)
+    self._active_container_type = None
+    self._container_types = []
+    self._active_cursor = None
+    self._add_active_container_method = None
+    self._store = redis_store.RedisStore(
+        definitions.STORAGE_TYPE_TASK,
+        session_identifier=task.session_identifier,
+        task_identifier=task.identifier)
+    self._store.Open(redis_client=redis_client)
+    self._event_data_identifier_mappings = {}
+    self._add_container_type_methods = {}
+    self._active_extra_containers = []
+
+    # Create a runtime lookup table for the add container type method. This
+    # prevents having to create a series of if-else checks for container types.
+    # The table is generated at runtime as there are no forward function
+    # declarations in Python.
+    for container_type, method_name in self._ADD_CONTAINER_TYPE_METHODS.items():
+      method = getattr(self, method_name, None)
+      if not method:
+        raise RuntimeError(
+            'Add method missing for container type: {0:s}'.format(
+                container_type))
+
+      self._add_container_type_methods[container_type] = method
+
+  def _AddAnalysisReport(self, analysis_report):
+    """Adds an analysis report.
+
+    Args:
+      analysis_report (AnalysisReport): analysis report.
+    """
+    self._storage_writer.AddAnalysisReport(analysis_report)
+
+  def _AddError(self, error):
+    """Adds an error.
+
+    Args:
+      error (ExtractionError): error.
+    """
+    self._storage_writer.AddError(error)
+
+  def _AddEventSource(self, event_source):
+    """Adds an event source.
+
+    Args:
+      event_source (EventSource): event source.F
+    """
+    self._storage_writer.AddEventSource(event_source)
+
+  def _AddEventTag(self, event_tag):
+    """Adds an event tag.
+
+    Args:
+      event_tag (EventTag): event tag.
+    """
+    self._storage_writer.AddEventTag(event_tag)
+
+  def _AddEvent(self, event):
+    """Adds an event.
+
+    Args:
+      event (EventObject): event.
+    """
+    if hasattr(event, 'event_data_row_identifier'):
+      event_data_identifier = identifiers.SQLTableIdentifier(
+          self._CONTAINER_TYPE_EVENT_DATA,
+          event.event_data_row_identifier)
+      lookup_key = event_data_identifier.CopyToString()
+
+      event_data_identifier = self._event_data_identifier_mappings[lookup_key]
+      event.SetEventDataIdentifier(event_data_identifier)
+
+    # TODO: add event identifier mappings for event tags.
```

Comment:
does this todo needs to be addressed before merge?

Reply:
hmm OK, we should take a look at this TODO (also in SQLite) as a follow up then.